nixops/backends/gce.py

```diff
@@ -769,10 +782,35 @@ def backup(self, defn, backup_id):
                                     .format(volume.name, self.machine_name)
                 })
 
+            # Apply labels to snapshot just created
+            self.wait_for_snapshot_initiated(snapshot_name)
+
+            self.log("updating labels of snapshot '{0}'".format(snapshot_name))
+            self.connect().connection.request(
+                '/global/snapshots/%s/setLabels' %(snapshot_name),
+                method = 'POST', data = {
+                    'labels': defn.labels,
+                    'labelFingerprint':
+                        self.connect().connection.request("/global/snapshots/{0}".format(snapshot_name), method='GET').object['labelFingerprint']
+            })
+
             backup[k] = snapshot_name
             _backups[backup_id] = backup
             self.backups = _backups
 
+    def wait_for_snapshot_initiated(self, snapshot_name):
+        while True:
+            try:
+                snapshot_status = self.connect().connection.request("/global/snapshots/{0}".format(snapshot_name), method='GET').object['status']
```

Comment:
No reason, that's actually better, I changed it that way and re-tested, thanks for pointing that out.

Also, would you think the same should be done here or that's okay (line 794) ?
`self.connect().connection.request("/global/snapshots/{0}".format(snapshot_name), method='GET').object['labelFingerprint']`

Reply:
right, maybe wait_for_snapshot_initiated() can return the object so that you re-use that to get the labelFingerprint instead of re-doing the api call.

Reply:
Actually it turns out the GCESnapshot object returned by libcloud doesn't have an attribute 'labelFingerprint'. Here's an example:
`<VolumeSnapshot id=542163224484327809 size=0 driver=Google Compute Engine state=None>`

I did also checked that here, the attributes being returned are in the function _to_snapshot(): http://libcloud.readthedocs.io/en/v0.14.0/_modules/libcloud/compute/drivers/gce.html

So I guess we're gonna need to use a raw http request then, no ? Thanks.

Reply:
yeah bummer